### PR TITLE
Remove globals from prepAsm

### PR DESCRIPTION
**Pull request by bstefanuk**
Removes globals from `prepAsm` in *TensileCreateLibrary* and adds type hinting for `writeKernels`

### CODE REVIEW

File: Tensile/BenchmarkProblems.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def writeBenchmarkFiles(stepBaseDir, solutions, problemSizes, \
     problemType = solutions[0]["ProblemType"]
     codeObjectFiles = writeKernels( \
             globalParameters["WorkingPath"], globalParameters["CxxCompiler"], \
-            [problemType], solutions, kernels, kernelHelperOjbs, \
+            globalParameters, solutions, kernels, kernelHelperOjbs, \
```
Comment (ellosel) [inline Tensile/BenchmarkProblems.py:153]: Are we passing in the global params now?

File: Tensile/TensileCreateLibrary.py [hunk 2]
```diff
@@ -396,69 +397,78 @@ def buildSourceCodeObjectFiles(CxxCompiler, kernelFiles, outputPath):
 
 
 ################################################################################
-def prepAsm(kernelWriterAssembly):
-    """
-    Create and prepare the assembly directory  - called ONCE per output dir:
+def prepAsm(kernelWriterAssembly: KernelWriterAssembly, buildPath: Path):
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:400]: Should we test this function as a part of this PR?

File: Tensile/TensileCreateLibrary.py [hunk 3]
```diff
@@ -396,69 +397,78 @@ def buildSourceCodeObjectFiles(CxxCompiler, kernelFiles, outputPath):
...
+
+    Args:
+        kernelWriterAssembly: Assembly writer object.
+        buildPath: Path to directory where assembly files will be written.
     """
-    asmPath = ensurePath(os.path.join(globalParameters["WorkingPath"], "assembly"))
+    asmPath = buildPath / "assembly"
+    asmPath.mkdir(exist_ok=True)
+
     isa = globalParameters["CurrentISA"]
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:414]: What about this global?

File: Tensile/TensileCreateLibrary.py (hunk 2 above)
Comment (bstefanuk) [reply Tensile/TensileCreateLibrary.py:400]: Yes, I will write some tests and add them.

File: Tensile/BenchmarkProblems.py (hunk 1 above)
Comment (bstefanuk) [reply Tensile/BenchmarkProblems.py:153]: The previous parameter `List[problemType]` was unused in the function. I replaced it with _params_, which, in TensileCreateLibrary main is just the _args_ dictionary. Elsewhere in the code, I replaced it with global parameters to appease the tests. This is primarily a stop-gap while we continue to migrate away from dependence on global parameters.

File: Tensile/TensileCreateLibrary.py (hunk 3 above)
Comment (bstefanuk) [reply Tensile/TensileCreateLibrary.py:414]: Good catch, was focused on the "WorkingPath" parameter and had blinders on.

File: Tensile/BenchmarkProblems.py (hunk 1 above)
Comment (bstefanuk) [reply Tensile/BenchmarkProblems.py:153]: Alternatively, I remove this parameter altogether and we rely on the global parameters in **writeKernels**

**Review by ellosel — APPROVED** [review]

**Pull request merged**